### DISCUSSION TITLE
Tab-complete description of top-level command

### DISCUSSION THREAD

**Discussion by jdholtz** [fish-shell/fish-shell#9805] [General]
Hello. I am creating a Fish completion script for my own command. I want to add a description for my command when a list of options comes up, but I am unsure how to do this. Currently, the description just says `(command)`. 

I already know I can use the `--description` flag to add a description to subcommands, but I want to do this for my top-level command. I tried to do `complete -c $progname -d "<description here>"` and looked at the `share/completions` directory in this repository but I could not find out how to do it.

Here is a screenshot displaying my 'custom-command'. You can see that most of the other commands' descriptions have an actual description, but my 'custom-command' just has `(command)`
![2023-05-21-185752_1573x74_scrot](https://github.com/fish-shell/fish-shell/assets/89617856/27fcfb55-1080-436b-a497-c5ac4db2b117)

How can I add a custom description for this command. Thanks!

**Comment by faho** [comment]
This description is either taken from the man page or, for functions, from the `--description` value.

For example if you define

```fish
function custom_command --description "This is my custom command"
    echo Hello I am custom_command
end
```

You'll see it described as "This is my custom command" - but only once it's loaded, so if it's in a file in ~/.config/fish/functions/ it won't appear until you run it once.

That's also why the completions won't work - before you try to complete the command, i.e. do `custom_command <TAB>` those aren't loaded.

**Reply by jdholtz** [reply]
Thanks for the response. 

> This description is either taken from the man page

My command does have a man page. Would there be any reason fish would not take the description from the man page? [Here](https://github.com/jdholtz/pacman-venv/blob/master/doc/pacman-venv.8) is a link to my project's man page for reference.

**Reply by faho** [reply]
Is the man page installed in your man path? Is it listed in the output of `__fish_describe_command`? Is the command called by the same name as the manpage? Is it listed when you run `apropos "^pacman-venv"`?

**Reply by jdholtz** [reply]
Yes, it is installed at `/usr/share/man/man8`. Those two commands did not output anything with my custom command. What is strange is I can do `man pacman-venv` and it shows up. It also has the same exact man path as other packages that show a description (I used `whereis pacman-venv` to see where the man page is located). 

I was able to reproduce the issue on another computer with a non-fish environment, so it definitely is not an issue with fish. Thanks for helping me out.

**Reply by faho** [reply]
See that you've updated the mandb database after you installed it - typically this is done via man-db.timer (which starts man-db.service), or you can manually run `mandb` as root.

**Reply by jdholtz** [reply]
Running mandb as root fixed it. Thanks! Previously I only tried running it as my user which makes sense why it wasn’t working correctly.
